src/extension/gutters.ts

```diff
@@ -76,6 +76,15 @@ export class Gutters {
         }
     }
 
+    public async toggleCoverageForActiveFile() {
+        try {
+            await this.coverageService.toggleCoverage();
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        } catch (error: any) {
+            this.handleError("toggleCoverageForActiveFile", error);
+        }
+    }
```

Comment:
Looking pretty good overall, we might want an integration test to confirm that the toggling works how we expect (see link below to the current ones we have in the test suite).

I might also have some free time today to add that test coverage before we merge this 😁.
https://github.com/ryanluker/vscode-coverage-gutters/blob/master/test/extension.test.ts#L184-L205

Reply:
Yes. I looked into adding an integration test a little bit but wasn't quite able to figure it out. Be great if you could do the honours - I'll follow along.

Reply:
@anishkny I pushed a test for us to use, I also tested out the feature locally and I think it is working well!
There is a bit of oddity with how it interacts with the watch command (it removes the coverage from the current page but doesn't disable the watching... maybe we want it like that?)

Reply:
Oopps I can't push to your branch in your fork... I pushed the commit here for you to grab when you get a chance.
https://github.com/anishkny/vscode-coverage-gutters/pull/1

Reply:
Thanks for adding the test! Merged into this PR.

Passing:

https://github.com/anishkny/vscode-coverage-gutters/actions/runs/3511153332/jobs/5882576842#step:6:108

```
✔ Run toggle coverage on python test file x2 @integration (77ms)
```